src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -1,59 +1,97 @@
 import com.googlecode.objectify.cmd.Query;
 
+import teammates.client.connector.DatastoreClient;
+import teammates.client.util.ClientProperties;
+import teammates.common.util.Const;
+import teammates.common.util.HibernateUtil;
 import teammates.storage.entity.Course;
 import teammates.storage.entity.CourseStudent;
+import teammates.storage.sqlentity.BaseEntity;
 import teammates.storage.sqlentity.Section;
 import teammates.storage.sqlentity.Student;
 import teammates.storage.sqlentity.Team;
+import teammates.test.FileHelper;
 
 /**
  * Data migration class for course entity.
  */
 @SuppressWarnings("PMD")
-public class DataMigrationForCourseEntitySql extends
-        DataMigrationEntitiesBaseScriptSql<teammates.storage.entity.Course, teammates.storage.sqlentity.BaseEntity> {
+public class DataMigrationForCourseEntitySql extends DatastoreClient {
+
+    private static final String BASE_LOG_URI = "src/client/java/teammates/client/scripts/log/";
+
+    private static final int BATCH_SIZE = 1000;
+
+    private List<BaseEntity> entitiesSavingBuffer;
+
+    // Creates the folder that will contain the stored log.
+    static {
+        new File(BASE_LOG_URI).mkdir();
+    }
+
+    AtomicLong numberOfAffectedEntities;
+    AtomicLong numberOfScannedKey;
+    AtomicLong numberOfUpdatedEntities;
+
+    public DataMigrationForCourseEntitySql() {
+        numberOfAffectedEntities = new AtomicLong();
+        numberOfScannedKey = new AtomicLong();
+        numberOfUpdatedEntities = new AtomicLong();
+
+        entitiesSavingBuffer = new ArrayList<>();
+
+        String connectionUrl = ClientProperties.SCRIPT_API_URL;
+        String username = ClientProperties.SCRIPT_API_NAME;
+        String password = ClientProperties.SCRIPT_API_PASSWORD;
+
+        HibernateUtil.buildSessionFactory(connectionUrl, username, password);
+    }
 
     public static void main(String[] args) {
         new DataMigrationForCourseEntitySql().doOperationRemotely();
     }
 
-    @Override
     protected Query<Course> getFilterQuery() {
         return ofy().load().type(teammates.storage.entity.Course.class);
     }
 
-    @Override
     protected boolean isPreview() {
         return false;
     }
 
     /*
      * Sets the migration criteria used in isMigrationNeeded.
      */
-    @Override
     protected void setMigrationCriteria() {
         // No migration criteria currently needed.
     }
 
-    @Override
     protected boolean isMigrationNeeded(Course entity) {
         return true;
     }
 
-    @Override
     protected void migrateEntity(Course oldCourse) throws Exception {
         teammates.storage.sqlentity.Course newCourse = createCourse(oldCourse);
-        // TODO: add shutdown hook to save the entity
-        // Runnable shutdownScript = () -> { cascadeDelete(newCourse)};
-        // Runtime.getRuntime().addShutdownHook(new Thread(shutdownScript));
 
         migrateCourseEntity(newCourse);
         // verifyCourseEntity(newCourse);
```

Comment:
After we migrate the course we should ensure that the buffer is flushed before marking the course as migrated in case of any errors